src/components/navBar/navBar.js

```diff
@@ -0,0 +1,400 @@
+/**
+ * @ngdoc module
+ * @name material.components.navBar
+ */
+
+
+angular.module('material.components.navBar', ['material.core'])
+    .controller('MdNavBarController', MdNavBarController)
+    .directive('mdNavBar', MdNavBar)
+    .controller('MdNavLinkController', MdNavLinkController)
+    .directive('mdNavLink', MdNavLink);
+
+
+/*****************************************************************************
+ *                            PUBLIC DOCUMENTATION                           *
+ *****************************************************************************/
+
+
+/*****************************************************************************
+ *                                IMPLEMENTATION                             *
+ *****************************************************************************/
+
+function MdNavBar() {
+  return {
+    restrict: 'E',
+    transclude: true,
+    controller: MdNavBarController,
+    controllerAs: 'ctrl',
+    bindToController: true,
+    scope: {'selectedTabName': '=?'},
+    template: '<div class="suite md-nav-bar">' +
+        '<nav role="navigation">' +
+        '<ul class="md-tab-container" layout="row" ng-transclude role="listbox"' +
+        'tabindex="0"' +
+        'ng-focus="ctrl.onFocus()"' +
+        'ng-blur="ctrl.onBlur()"' +
+        'ng-keydown="ctrl.onKeydown($event)"' +
+        'aria-label="[[Top-level navigation links|' +
+        'Description of navigation links, which provide quick access to' +
+        'main sections of the site.]]">' +
+        '</ul>' +
+        '</nav>' +
+        '<div class="md-ink-bar"></div>' +
+        '</div>'
+  };
+}
+
+/**
+ * Controller for the nav-bar component.
+ *
+ * Accessibility functionality is implemented as a site navigator with a
+ * listbox, according to
+ * https://www.w3.org/TR/wai-aria-practices/#Site_Navigator_Tabbed_Style
+ * @param {!suiteTypes.jQuery} $element
+ * @param {!angular.Scope} $scope
+ * @ngInject
```

Comment:
Add `@constructor`
(WebStorm understands JsDoc annotations like `@constructor` and has some intelligence around them)